posts/composite-images/making-composite-images.qmd

```diff
@@ -0,0 +1,246 @@
+  ```
+
+- `8bit_autocontrast` – Images in the 8bit folder are autocontrasted 
+based on the user supplied clipping value. The file format will follow 
+fov_num>_ch<#>_8bit_autocontrast.<user_format>
+
+  Example
+  ```
+  F001_ch0_8bit_autocontrast.png (for user_format = png)
+  ```
+
+- `composite` - Composite images created from the images in the 
+`8bit` folder. The composite type is a screen composite. 
+The file format will follow fov_num>_composite.<user_format>
+
+  Example
+  ```
+  F001_composite.jpg (for user_format = jpg)
+  ```
+
+- `composite_autocontrast` -  Images in the composite_folder are 
+autocontrasted based on the user specified clipping value. 
+The file format will follow fov_num>_composite_autocontrast.<user_format>
+
+  Example
+  ```
+  F001_composite_autocontrast.png (for user_format = png)
+  ```
+
+# Usage
+
+```
+navigate to your Morphology2D folder
+python /path/to/your/make_composite.py # and follow the on-screen prompts
+```
+
+Regex pattern matching on 2D morphology file name format is implemented, 
+however, only NanoString 2D morphology files should be present in the 
+folder containing the make_composite script. 
+
+# Example
+
+The example dataset that we used was the mouse coronal hemisphere
+FFPE that is available to download from NanoString's website [here](https://nanostring.com/products/cosmx-spatial-molecular-imager/ffpe-dataset/cosmx-smi-mouse-brain-ffpe-dataset/){target="blank"}. 
+
+The Morphology2D folder is found within the CellStatsDir folder and has TIF files
+for each of the 130 FOVs.
+
+```{default filename="Terminal"}
+#| eval: false
+cd /path/to/slide/CellStatsDir/Morphology2D
+```
+
+```{default filename="Terminal"}
+#| eval: false
+tree -L 1
+```
+
+```
+├── 20230406_205644_S1_C902_P99_N99_F001.TIF
+├── 20230406_205644_S1_C902_P99_N99_F002.TIF
+...
+├── 20230406_205644_S1_C902_P99_N99_F129.TIF
+└── 20230406_205644_S1_C902_P99_N99_F130.TIF
+
+```
+
+Once in the Morphology2D folder, simply run the script and follow the on-screen
+prompts (@fig-run).
+
+```{default filename="Terminal"}
+#| eval: false
+python /path/to/CosMx-Analysis-Scratch-Space/assets/make-composite/make_composite.py
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-run"
+#| fig-cap: "Screenshot of standard output from terminal following script execution. In this example, I set the clipping percentage to 3 and the output to `png`. On an Macbook Pro M1, this took about 25 minutes to process 130 FOVs."
+
+knitr::include_graphics("./figures/fig-out.png")
+```
+
+When complete, the structure of the Morphology2D folder should resemble this:
+
+```{default filename="Terminal"}
+#| eval: false
+tree -L 1
+```
+
+```
+├── 20230406_205644_S1_C902_P99_N99_F001.TIF
+├── 20230406_205644_S1_C902_P99_N99_F002.TIF
+...
+├── 20230406_205644_S1_C902_P99_N99_F129.TIF
+├── 20230406_205644_S1_C902_P99_N99_F130.TIF
+├── 8bit
+├── 8bit_autocontrast
+├── composite
+├── composite_autocontrast
+└── raw
+```
+
```

Comment:
Can you add a concluding sentence about what I should do next? A sentence or two, similar to this: These composite images can now be imported into open-source software or explored further.